File: src/main/java/com/efedotov/vaultly/dto/user/UserProfileDto.java
package com.efedotov.vaultly.dto.user;

import java.util.UUID;

import lombok.Data;

@Data
public class UserProfileDto {
    private UUID id;
    private String email;
    private String username;
    private String firstName;
    private String lastName;
    private String avatarUrl;
    private Long storageUsed;
    private Long storageLimit;
    private String publicKey;
}

File: lib/server/model/user/user_profile_dto/user_profile_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'user_profile_dto.freezed.dart';
part 'user_profile_dto.g.dart';

@freezed
abstract class UserProfileDto with _$UserProfileDto {
  const factory UserProfileDto({
    String? id,
    String? email,
    String? username,
    String? firstName,
    String? lastName,
    String? avatarUrl,
    int? storageUsed,
    int? storageLimit,
    String? publicKey,
    final bool? totpEnabled,
  }) = _UserProfileDto;

  factory UserProfileDto.fromJson(Map<String, dynamic> json) =>
      _$UserProfileDtoFromJson(json);
}
